src/transform.js

```diff
@@ -14,13 +14,13 @@ module.exports = async function(source, options) {
 
   if (!result) return null;
 
-  const { code, map, metadata } = result;
+  const { map } = result;
 
   if (map && (!map.sourcesContent || !map.sourcesContent.length)) {
     map.sourcesContent = [source];
   }
 
-  return { code, map, metadata };
+  return result;
```

Comment:
My one concern here is that `result.options` is not serializable to JSON. This change makes the full `options` object available to `result` override callbacks, but it'll only be useful if you're running without the cache, since the cache version would reload and be missing stuff.

When you asked originally I was fine with removing the whitelist of props, but now I'm leaning toward saying we'd be better to at least blacklist the `result.options` property.

Reply:
Good point. I can alternatively just add `ast` to the whitelist if that's better?

Reply:
The full set of props is at https://github.com/babel/babel/blob/9283efaeb7a8c5b505b19332c3407dc44c668777/packages/babel-core/src/transformation/index.js#L64

If you want to expand the whitelist to include all of those except `options` that sounds good to me.

Reply:
Will do! 🎉